File: e_learning_backend/src/main/java/com/ai/e_learning/service/CertificateService.java
package com.ai.e_learning.service;

import com.ai.e_learning.dto.CertificateDto;
import org.springframework.stereotype.Service;

@Service
public interface CertificateService {
    CertificateDto saveCertificate(CertificateDto certificateDto);
    CertificateDto getCertificateById(Long id);
    CertificateDto getCertificateByUserId(Long userId);

    CertificateDto getCertificateByUserIdAndCourseId(Long userId, Long courseId);

    boolean checkUserCertificate(Long userId, Long courseId);
}

File: e_learning_frontend/src/app/components/services/certificate.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Certificate } from '../models/certificate.model';

@Injectable({
  providedIn: 'root'
})
export class CertificateService {
  private baseURL = "http://localhost:8080/api/certificate";
  constructor(private http: HttpClient) { }

  getCertificateById(id: number): Observable<Certificate> {
    return this.http.get<Certificate>(`${this.baseURL}/${id}`);
  }
  saveCertificate(certificate: Certificate): Observable<Certificate> {
    return this.http.post<Certificate>(this.baseURL, certificate);
  }
  getCertificateByUserId(id: number): Observable<Certificate> {
    return this.http.get<Certificate>(`${this.baseURL}/user/${id}`);
  }

  getCertificateByUserIdAndCourseId(userId: number, courseId: number): Observable<Certificate> {
    return this.http.get<Certificate>(`${this.baseURL}/user/${userId}/course/${courseId}`);
  }

  checkUserCertificate(userId: number, courseId: number): Observable<boolean> {
    return this.http.get<boolean>(`${this.baseURL}/checkUserCertificate`, {
      params: { userId: userId.toString(),courseId: courseId.toString() }
    });
  }
}
